spec/lib/declare_schema/prepare_testapp.rb

```diff
@@ -18,6 +18,8 @@
 require 'rails/generators'
 Rails::Generators.configure!(Rails.application.config.generators)
 
+ActiveRecord::Base.connection.schema_cache.clear!
```

Comment:
It took me more than an hour to figure out that this `schema_cache` was messing up tests by carrying state across from one test to the next.